rust/datafusion/src/logicalplan.rs

```diff
@@ -119,22 +113,24 @@ pub enum ScalarValue {
 }
 
 impl ScalarValue {
-    pub fn get_datatype(&self) -> DataType {
+    pub fn get_datatype(&self) -> Result<DataType> {
```

Comment:
Aside: It would be great if we could evolve `ScalarValue` in future releases, to behave like an `arrow::Array` that always has a value of 1. It would make situations like `computation(array, scalar {e.g. literal})` easier as our computation kernels would be able to handle literal values.
Wes worked on the C++ implementation a few weeks ago, I want to mimic it in the coming weeks.